File: src/main/java/me/centralhardware/znatoki/telegram/statistic/configuration/RateLimiterConfiguration.java
package me.centralhardware.znatoki.telegram.statistic.configuration;

import com.google.common.util.concurrent.RateLimiter;
import me.centralhardware.znatoki.telegram.statistic.Config;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@SuppressWarnings("UnstableApiUsage")
@Configuration
public class RateLimiterConfiguration {

    @Bean
    public RateLimiter getRateLimiter(){
        return RateLimiter.create(Config.Telegram.getTelegramRateLimit());
    }

}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/configuration/RateLimiterConfiguration.kt
package me.centralhardware.znatoki.telegram.statistic.configuration

import com.google.common.util.concurrent.RateLimiter
import me.centralhardware.znatoki.telegram.statistic.Config
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Suppress("UnstableApiUsage")
@Configuration
class RateLimiterConfiguration {

    @Bean
    fun getRateLimiter(): RateLimiter {
        return RateLimiter.create(Config.Telegram.rateLimit)
    }

}